web/routes/util.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+const fs = require( 'fs');
+
+const readDirectory = ( dir) => {
+  return new Promise( (resolve, reject) => {
+    fs.readdir( dir, (err, res) => {
+      if( err){
+        reject( err);
+      } else {
+        resolve( res);
+      }
+    });
+  });
+};
+
+module.exports = {
+  getDirPicCount: async function( dir) {
+    let image_count = -1;
+    try {
+      const listing = await readDirectory( dir);
+      image_count = listing.length;
+    } catch( e) {
+      if( e.code === "ENOENT") {
+        console.log( "creating missing directory:", dir)
+        // create missing directory
+        try {
+          fs.mkdirSync( dir, 484);
+          image_count = 0;
+        } catch( e) {
+          // we can't get error EEXIST here so bail
+          console.error( "mkdirSync failed:", e);
+        }
+      } else {
+        console.error( "directory count failed:", e);
+      }
+    }
+    return image_count;
+  },
+  makeUrlsFromBlobs: function( parent_id, current_image_count, blobs) {
```

Comment:
Please give default values where necessary or check for type error in function body, specially for public methods.